src/main.js

```diff
@@ -19,31 +19,63 @@ userNewBody.addEventListener('input', verifyForm);
 
 ideaGallery.addEventListener('click', function(event) {
   event.preventDefault();
-  var activeStarBtn = document.querySelector('.star-active');
-  var inactiveStarBtn = document.querySelector('.star-inactive');
 
   if (event.target.className === 'delete') {
     deleteFromArray(event);
     deleteFromDOM(event);
   }
 
-  if (event.target.className === 'star-active') {
-    inactiveStarBtn.classList.remove('hide');
-    activeStarBtn.classList.add('hide');
-  }
-
   if (event.target.className === 'star-inactive') {
-    inactiveStarBtn.classList.add('hide');
-    activeStarBtn.classList.remove('hide');
+    toggleStarOnData(event);
+    toggleStarOnDOM(event);
+    console.log(savedIdeas);
+  } else if (event.target.className === 'star-active') {
+    toggleStarOffData(event);
+    toggleStarOffDOM(event);
+    console.log(savedIdeas);
   }
 })
 
+function toggleStarOnData() {
+  var clickedStar = event.target.closest('.box');
+  var starIdNumber = parseInt(clickedStar.id);
+
+  for (var i = 0; i < savedIdeas.length; i++) {
+    if (starIdNumber === savedIdeas[i].id) {
+      savedIdeas[i].star = true;
+    }
+  }
+}
+
+function toggleStarOnDOM() {
+  var starImg = event.target;
+  starImg.src = "assets/star-active.svg";
+  starImg.className = "star-active";
+}
```

Comment:
The event targets .src + className are changed here, which effects what's displayed.